app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -65,28 +65,21 @@ class SyncIndicatorViewController @Inject constructor(
     return showSyncIndicatorState.mergeWith(showDefaultSyncIndicatorState)
   }
 
-  private fun syncedFailureState(timestamp: Instant?): SyncIndicatorState {
+  private fun syncIndicatorState(timestamp: Instant?, maxIntervalSinceLastSync: Duration): SyncIndicatorState {
     if (timestamp == null) {
       return SyncPending
     }
 
-    return syncIndicatorState(timestamp)
-  }
-
-  private fun syncIndicatorState(timestamp: Instant?): SyncIndicatorState {
     val now = Instant.now(utcClock)
     val timeSinceLastSync = Duration.between(timestamp, now)
-
-    val maxIntervalSinceLastSync = Duration.ofHours(12)
     val mostFrequentSyncInterval = enumValues<SyncInterval>()
         .map { it.frequency }
         .min()!!
-
     return when {
       timeSinceLastSync > maxIntervalSinceLastSync -> ConnectToSync
       timeSinceLastSync > mostFrequentSyncInterval -> SyncPending
       timeSinceLastSync.isNegative -> SyncPending
```

Comment:
It's a bit confusing to read that `timeSinceLastSync` can be negative until you realize that this can happen if the user manually changes the device time. In fact, it looks baffling on the first look. :P 

Can this be made more obvious? It could be as simple as just extracting this to a variable and documenting the code by the variable's name:

```kotlin
val syncHappenedInTheFuture = timestamp > now

return when {
  syncHappenedInTheFuture -> SyncPending
  timeSinceLastSync > syncFailureThreshold -> ConnectToSync
  timeSinceLastSync > mostFrequentSyncInterval -> SyncPending
  else -> Synced(timeSinceLastSync.toMinutes())
}
```

What do you think?

Reply:
Makes sense. Will make this change.